operators/volume_loader/nrrd_loader.cpp

```diff
@@ -148,56 +148,96 @@ bool load_nrrd(const std::string& file_name, Volume& volume) {
-  if (!file.is_open()) {
-    holoscan::log_error("NRRD could not open {}", data_file_name);
-    return false;
-  }
-  const std::streampos file_size = file.tellg();
-  file.seekg(0, std::ios_base::beg);
-
-  if (compressed) {
-    // need to uncompress, first read to 'compressed_data' vector and then uncompress to 'data'
-    std::vector<uint8_t> compressed_data(file_size);
-
-    // read
-    file.read(reinterpret_cast<char*>(compressed_data.data()), compressed_data.size());
-
-    // uncompress
-    z_stream strm{};
-    int result = inflateInit2(&strm, 32 + MAX_WBITS);
-    if (result != Z_OK) {
-      holoscan::log_error("NRRD failed to uncompress {}, inflateInit2 failed with error code {}",
-                          data_file_name,
-                          result);
+    file.read(reinterpret_cast<char*>(data.get()), data_size);
+  } else if (data_file_name.size() != 0) {
+    std::ifstream file;
+
+    file.open(data_file_name, std::ios::in | std::ios::binary | std::ios::ate);
+    if (!file.is_open()) {
+      holoscan::log_error("NRRD could not open {}", data_file_name);
       return false;
     }
+    const std::streampos file_size = file.tellg();
+    file.seekg(0, std::ios_base::beg);
+    if (compressed) {
+      // need to uncompress, first read to 'compressed_data' vector and then uncompress to 'data'
+      std::vector<uint8_t> compressed_data(file_size);
 
-    strm.next_in = compressed_data.data();
-    strm.avail_in = compressed_data.size();
-    strm.next_out = data.get();
-    strm.avail_out = data_size;
-
-    result = inflate(&strm, Z_FINISH);
-    inflateEnd(&strm);
-    if (result != Z_STREAM_END) {
-      holoscan::log_error("NRRD failed to uncompress {}, inflate failed with error code {}",
-                          data_file_name,
-                          result);
-      return false;
+      // read
+      file.read(reinterpret_cast<char*>(compressed_data.data()), compressed_data.size());
+
+      // uncompress
+      z_stream strm{};
+      int result = inflateInit2(&strm, 32 + MAX_WBITS);
+      if (result != Z_OK) {
+        holoscan::log_error("NRRD failed to uncompress {}, inflateInit2 failed with error code {}",
+                            data_file_name,
+                            result);
+        return false;
+      }
+
+      strm.next_in = compressed_data.data();
+      strm.avail_in = compressed_data.size();
+      strm.next_out = data.get();
+      strm.avail_out = data_size;
+
+      result = inflate(&strm, Z_FINISH);
+      inflateEnd(&strm);
+      if (result != Z_STREAM_END) {
+        holoscan::log_error("NRRD failed to uncompress {}, inflate failed with error code {}",
+                            data_file_name,
+                            result);
+        return false;
+      }
+    } else {
+      holoscan::log_error("NRRD unsupported file format");
     }
-  } else {
-    file.read(reinterpret_cast<char*>(data.get()), data_size);
```

Comment:
I think this path for uncompressed data is missing from the new code.

Reply:
Thank you.